podium-gateway/src/main/java/nl/thehyve/podium/service/RequestFileService.java

```diff
@@ -0,0 +1,156 @@
+//                allowedToAdd = true;
+//            }
+//        }
+//
+//        // Reviewer can add files in Review status
+//        if(securityService.isCurrentUserInAnyOrganisationRole(organisationUuids, AuthorityConstants.REVIEWER)){
+//            if(requestStatus == RequestStatus.Review){
+//                allowedToAdd = true;
+//            }
+//        }
+//
+//        if(!allowedToAdd){
+//            throw ActionNotAllowed.forStatus(request.getStatus());
+//        }
+        try{
+            String uploadFolder = uploadDir + System.currentTimeMillis() + "/";
+            byte[] bytes = file.getBytes();
+            String pathString = uploadFolder + file.getOriginalFilename();
+            Path path = Paths.get(pathString);
+            File posFile = new File(path.toString());
+            File neededDir = new File(uploadFolder);
+
+            // Add the required folder(s) if it doesn't exist yet
+            if(!neededDir.exists()){
+                neededDir.mkdirs();
+            }
+            // Doublecheck if this doesn't exist, otherwise call this function again to generate a new uploadFolder string.
+            if(!posFile.exists()){
+                Files.write(path, bytes);
+                requestFile.setFileLocation(path.toString());
+            } else {
+                return this.addFile(owner, requestUuid, file, requestFileType);
+            }
+        } catch (IOException e) {
+            log.error("Exception saving File", e);
+        }
+
+        requestFileRepository.save(requestFile);
+
+        return requestFileMapper.processingRequestFileDtoToRequestFile(requestFile);
+    }
+
+    public ByteArrayResource getFile(IdentifiableUser requester, UUID requestUUID, UUID fileUuid) throws IOException{
+        RequestFile requestFile = requestFileRepository.findOneByUuidAndDeletedFalse(fileUuid);
+
+        Path path = Paths.get(requestFile.getFileLocation());
+        return new ByteArrayResource(Files.readAllBytes(path));
+    }
+
+    public List<RequestFileRepresentation> getFilesForRequest(IdentifiableUser requester, UUID requestUUID){
+        Request request = requestRepository.findOneByUuid(requestUUID);
+
+        List<RequestFile> files = requestFileRepository.findDistinctByRequestAndDeletedFalse(request);
+
+        List<RequestFileRepresentation> representations = new ArrayList<RequestFileRepresentation>();
+        for(RequestFile file : files){
+            RequestFileRepresentation representation = requestFileMapper.processingRequestFileDtoToRequestFile(file);
+            representations.add(representation);
+        }
+
+        return representations;
+    }
+
+    public Boolean deleteFile(IdentifiableUser requester, UUID fileUuid){
+        RequestFile requestFile = requestFileRepository.findOneByUuidAndDeletedFalse(fileUuid);
+
+        //Only owners can delete files.
+        if(requestFile.getOwner().equals(requester.getUserUuid())){
+            requestFile.setDeleted(true);
```

Comment:
Probably you also want to really delete the file. Consider the case when uploaded files contain sensitive data, for instance (and should not have been uploaded).